docs/clustering.md

```diff
@@ -18,29 +18,36 @@ that the other libraries boast.
 from sklearn.manifold import TSNE
 from turftopic import ClusteringTopicModel
 
-model = ClusteringTopicModel(clustering=TSNE())
+model = ClusteringTopicModel(dimensionality_reduction=TSNE())
 ```
 
 It is common practice to reduce the dimensionality of the embeddings before clustering them.
 This is to avoid the curse of dimensionality, an issue, which many clustering models are affected by.
-Dimensionality reduction by default is done with scikit-learn's **TSNE** implementation in Turftopic,
+Dimensionality reduction by default is done with **TSNE** in Turftopic,
 but users are free to specify the model that will be used for dimensionality reduction.
 
+!!! tip "Use openTSNE for better performance!"
+    By default, a scikit-learn implementation is used, but if you have the openTSNE package installed on your system, Turftopic will automatically use it.
+    You can potentially speed up your clustering topic models by multiple orders of magnitude.
+    ```bash
+    pip install opentsne
+    ```
+
 ??? note "What reduction model should I choose?"
     Our knowledge about the impacts of choice of dimensionality reduction is limited, and has not yet been explored in the literature.
     Top2Vec and BERTopic both use UMAP, which has a number of desirable properties over alternatives (arranging data points into cluster-like structures, better preservation of global structure than TSNE, speed).
```

Comment:
Okay, so the rationale behind this is mainly that I want to keep the package's dependencies to a minimum. Especially anything built with Numba (like UMAP) I would like to keep a mile's distance from.
This is simply because Numba regularly has a hard time catching up with other package's versions, mainly Numpy, and it would regularly cause headaches for people to get dependency conflicts when installing our package.

The way Turftopic is now, people don't have to install anything, because most of it is based on packages, people very likely already have, and I would also love to maintain permissiveness in versions.

I did, however want to add an alternative, that would be faster than sklearn's TSNE implementation, without a major shift in behaviour, this is why the package automatically uses openTSNE when it's installed.

If a good implementation of UMAP comes around, that doesn't have the same headaches as the current one, we can totally switch to that.

Reply:
Might be worth noting that in the message (we do not use UMAP as a default to keep the package minimal)